src/main/java/io/vertx/httpproxy/spi/cache/Resource.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.httpproxy.spi.cache;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.shareddata.ClusterSerializable;
+import io.vertx.httpproxy.Body;
+import io.vertx.httpproxy.ProxyResponse;
+import io.vertx.httpproxy.impl.ParseUtils;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+
+/**
+ * The cached object.
+ */
+public class Resource implements ClusterSerializable {
+
+  private static final Charset UTF_8 = StandardCharsets.UTF_8;
+
+  private String absoluteUri;
+  private int statusCode;
+  private String statusMessage;
+  private MultiMap headers;
+  private long timestamp;
+  private long maxAge;
+  private Instant lastModified;
+  private String etag;
+  private Buffer content = Buffer.buffer();
+
+  // For serialization purposes, do not remove.
+  public Resource() {
+  }
+
+  public Resource(String absoluteUri, int statusCode, String statusMessage, MultiMap headers, long timestamp, long maxAge) {
+    String lastModifiedHeader = headers.get(HttpHeaders.LAST_MODIFIED);
+    this.absoluteUri = absoluteUri;
+    this.statusCode = statusCode;
+    this.statusMessage = statusMessage;
+    this.headers = headers;
+    this.timestamp = timestamp;
+    this.maxAge = maxAge;
+    this.lastModified = lastModifiedHeader != null ? ParseUtils.parseHeaderDate(lastModifiedHeader) : null;
+    this.etag = headers.get(HttpHeaders.ETAG);
+  }
+
+  public void init(ProxyResponse proxyResponse) {
```

Comment:
this method should be moved outside of the Resource, the cache SPI should not care about the ProxyResponse API